src/main/java/io/fabric8/maven/docker/access/KeyStoreUtil.java

```diff
@@ -41,28 +43,40 @@ public static KeyStore createDockerKeyStore(String certPath) throws IOException,
         return keyStore;
     }
 
-    public static PrivateKey loadPrivateKey(String keyPath) throws IOException, GeneralSecurityException {
-        PEMKeyPair keyPair = loadPEM(keyPath);
-        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(keyPair.getPrivateKeyInfo().getEncoded());
-        return KeyFactory.getInstance("RSA").generatePrivate(keySpec);
+    static PrivateKey loadPrivateKey(String keyPath) throws IOException, GeneralSecurityException {
+        try (Reader reader = new FileReader(keyPath);
+            PEMParser parser = new PEMParser(reader)) {
+            Object readObject = parser.readObject();
+            if (readObject instanceof PEMKeyPair) {
+                PEMKeyPair keyPair = (PEMKeyPair) readObject;
+                return generatePrivateKey(keyPair.getPrivateKeyInfo());
+            } else if (readObject instanceof PrivateKeyInfo) {
+                return generatePrivateKey((PrivateKeyInfo) readObject);
+            }
+        }
+        return null;
```

Comment:
I would throw an exception in that case, since a private key is mandatory for the further processing. It better to stop where the issue happens and where we have most of the error context (i.e. read object type which is wrong).